sros2/test/sros2/commands/security/verbs/test_create_key.py

```diff
@@ -22,11 +23,29 @@
 from cryptography.hazmat.primitives import serialization
 from cryptography.hazmat.primitives.asymmetric import ec
 
-from ros2cli import cli
+import pytest
 
+from ros2cli import cli
 from sros2.api import create_keystore
 
 
+# This fixture will run once for the entire module (as opposed to once per test)
+@pytest.fixture(scope='module')
+def node_keys_dir(tmp_path_factory):
```

Comment:
How does this function get the argument `tmp_path_factory`?

I am asking because the nightly CI jobs fail the test: http://build.ros2.org/view/Eci/job/Eci__nightly-fastrtps_ubuntu_bionic_amd64/19/testReport/sros2.test.sros2.commands.security.verbs/test_create_key/test_create_key/

The nightly job on ci.ros2.org seems to pass (https://ci.ros2.org/view/nightly/job/nightly_linux_debug/1254/testReport/sros2.test.sros2.commands.security.verbs/test_create_key/test_create_key/). Not sure what is different between those.

Reply:
Well that's an interesting failure. This is a [built-in pytest fixture](https://docs.pytest.org/en/latest/tmpdir.html). Any chance the version of pytest differs between those two tests?

Reply:
I faced multiple issues about mismatching versions of pytest recently. My guess is that one of these jobs installs a recent pytest from pip and the other one an older version from apt

Reply:
That is certainly the case: one uses the Debian package from Ubuntu Bionic, the other uses the latest release (currently we pinned 4.x instead of using 5 - but that is a different reason). So the code should work with either of the versions.

Reply:
Good theory, the one from apt may need to also install `python3-pytest-tempdir`.

Reply:
It will only do so if you declare it explicitly as a test dependency in the package manifest.

Reply:
> Good theory, the one from apt may need to also install python3-pytest-tempdir (total guess based on the name, but seems plausible).

:+1: that would make sense. To do so it will need to be listed as a test dependency in sros2's package.xml and added to the [rosdep database](https://github.com/ros/rosdistro/blob/master/rosdep/python.yaml) for rosdep to resolve it properly.

Reply:
Yeah, too bad they're split out like that :man_shrugging: . Let me confirm the theory and I'll take care of it if that happens to be the case.

Reply:
> is there a way to run the failing test on a pull request instead of the one that passes? 

The master branch (which will become Eloquent) doesn't have PR jobs yet. So there is no automated way to do that. If you make a PR we can manually trigger a job for it (http://build.ros2.org/view/Eci/job/Eci__overlay_ubuntu_bionic_amd64/).

> Can we do something to bring those testbeds more closely inline with each other?

Not easily - both systems are designed for very different cases and therefore intentionally using different dependencies (Debian packages vs. latest).

Reply:
Nope, I was wrong. `python3-pytest-tempdir` is not required. Apparently, despite it not being present in the changelog, `tmp_path_factory` wasn't introduced into v5.0.0 of pytest. I'll propose a fix using an older factory.

For what it's worth, it would help avoid these issues if the "install from source" guide installed Debian packages instead of pip. Otherwise issues like this will continue to crop up, with the CI being the limiting factor. Combined with the fact that these jobs don't seem to be the ones that are typically run on PRs, it means we'll continue to catch these issues only after they land, which isn't ideal.

Reply:
PR is up at #145.